pkg/labels/labels.go

```diff
@@ -35,8 +40,29 @@ func (s selectorOp) hasValue(val string) bool {
 	return false
 }
 
+func (s *selectorOp) containsValue(val string) bool {
+	for i := range s.values {
+		if strings.Contains(val, s.values[i]) {
+			return true
+		}
+	}
+	return false
+}
+
 func (s *selectorOp) match(labels Labels) bool {
 	val, exists := labels[s.key]
+	if s.key == K8sPodNamespace {
+		switch s.operator {
+		case opExists:
+			return s.containsValue(labels[K8sPodNamespace])
+		case opNotIn:
+			return !s.hasValue(labels[K8sPodNamespace])
+		case opIn:
+			return s.hasValue(labels[K8sPodNamespace])
+		default:
+			return false
+		}
+	}
```

Comment:
Is this still needed?

Reply:
Not sure I understand. Can you provide an example?

Reply:
I'm not sure I understand still.

Aren't we matching namespaces using `k8s:io.kubernetes.pod.namespace` as a key, and the namespace name as the value?

So it's not clear to me how the `Exists` operator can be used in this case.

If the user wants to match pods having the `cilium` namespace, wouldn't they used the Equal operator? That is `k8s:io.kubernetes.pod.namespace <Equals> cilium`

Reply:
suppose we have namespaces ```cilium``` ```ciliumXYZ``` ```tetragon``` my thought was 

```exists``` operator for value ```cilium``` should match ```cilium``` and ```ciliumXYZ```

```equal``` operator for value ```cilium``` will only match ```cilium``` .

If we leave that additional conditional in ```exists``` operator maybe it will check if ```k8s:io.kubernetes.pod.namespace``` is available in any pod which might not be the behaviour we want for this right ?

Reply:
> exists operator for value cilium should match cilium and ciliumXYZ
> equal operator for value cilium will only match cilium

Was the intention to use the exists operator as a substring match? I believe that for every
other label, the exists checks whether the key exists. I dont think it makes sense to change the semantics for this specific key.

> If we leave that additional conditional in exists operator maybe it will check if k8s:io.kubernetes.pod.namespace is available in any pod which might not be the behaviour we want for this right ?

Do you mean that the check for the exists operator would return true for every pod? I think that's fine.

Reply:
ohk got it thanks